### PR TITLE
Part of #11159 Added smart pointers to a few functions

### PR DESCRIPTION
**Pull request by sohiladwani**
Looking to get a code review if possible!

### CODE REVIEW

**Comment by sohiladwani** [comment]
Sorry for spam commits, im debugging locally and trying to find the problem. Sometimes its compiling and sometimes its not and on github actions its always failing. Some weird behavior im dealing with.

File: src/openrct2/CmdlineSprite.cpp [hunk 1]
```diff
@@ -105,7 +106,7 @@ static bool sprite_file_open(const utf8* path)
             return false;
         }
 
-        spriteFileData = static_cast<uint8_t*>(malloc(spriteFileHeader.total_size));
+        spriteFileData = std::make_unique<uint8_t>(spriteFileHeader.total_size).get();
```
Comment (duncanspumpkin) [inline src/openrct2/CmdlineSprite.cpp:109]: You allocate and immediately free your memory on this line. You need to save the result of this not discard it

File: src/openrct2/cmdline/BenchSpriteSort.cpp [hunk 2]
```diff
@@ -120,8 +120,7 @@ static std::vector<paint_session> extract_paint_session(const std::string parkFi
         dpi.width = resolutionWidth;
         dpi.height = resolutionHeight;
         dpi.pitch = 0;
-        dpi.bits = static_cast<uint8_t*>(malloc(dpi.width * dpi.height));
-
+        dpi.bits = std::make_unique<uint8_t>(dpi.width * dpi.height).get();
```
Comment (duncanspumpkin) [inline src/openrct2/cmdline/BenchSpriteSort.cpp:123]: Allocates and then frees immediately

File: src/openrct2-win/openrct2-win.cpp [hunk 3]
```diff
@@ -39,16 +40,16 @@ int wmain(int argc, wchar_t** argvW, [[maybe_unused]] wchar_t* envp)
...
     return exitCode;
 }
+static std::unique_ptr<char* []> GetCommandLineArgs(int argc, wchar_t** argvW)
 
-static char** GetCommandLineArgs(int argc, wchar_t** argvW)
 {
     // Allocate UTF-8 strings
-    auto argv = static_cast<char**>(malloc(argc * sizeof(char*)));
+    auto argv = std::make_unique<char*[]>(argc);
+
```
Comment (duncanspumpkin) [inline src/openrct2-win/openrct2-win.cpp:52]: This is going to leak memory

File: src/openrct2/network/Network.cpp [hunk 4]
```diff
@@ -1646,7 +1646,7 @@ uint8_t* Network::save_for_network(size_t& out_size, const std::vector<const Obj
     else
     {
         log_warning("Failed to compress the data, falling back to non-compressed sv6.");
-        header = static_cast<uint8_t*>(malloc(size));
+        header = std::make_unique<uint8_t>(size).get();
```
Comment (duncanspumpkin) [inline src/openrct2/network/Network.cpp:1649]: You have set header to freed memory again

File: src/openrct2/platform/Windows.cpp [hunk 5]
```diff
@@ -209,7 +209,8 @@ bool platform_get_steam_path(utf8* outPath, size_t outSize)
         return false;
     }
 
-    wSteamPath = (wchar_t*)malloc(size);
+    wSteamPath = std::make_unique<wchar_t>(size).get();
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Windows.cpp:212]: Free d memory again

File: src/openrct2/util/SawyerCoding.cpp [hunk 6]
```diff
@@ -70,8 +70,10 @@ size_t sawyercoding_write_chunk_buffer(uint8_t* dst_file, const uint8_t* buffer,
             free(encode_buffer);
             break;
         case CHUNK_ENCODING_RLECOMPRESSED:
-            encode_buffer = static_cast<uint8_t*>(malloc(chunkHeader.length * 2));
-            encode_buffer2 = static_cast<uint8_t*>(malloc(0x600000));
+            // encode_buffer = static_cast<uint8_t*>(malloc(chunkHeader.length * 2));
+            encode_buffer = std::make_unique<uint8_t>(chunkHeader.length * 2).get();
+            // encode_buffer2 = static_cast<uint8_t*>(malloc(0x600000));
+            encode_buffer2 = std::make_unique<uint8_t>(0x600000).get();
```
Comment (duncanspumpkin) [inline src/openrct2/util/SawyerCoding.cpp:76]: And again

**Comment by tupaschoal** [comment]
Seems like you merged instead of rebase, so it's kind of hard to review.

Please rebase instead and keep only your changes, this explains the difference: https://www.atlassian.com/git/tutorials/merging-vs-rebasing

**Comment by tupaschoal** [comment]
As this hasn't been touched in a while, I'm closing this PR. Please reopen if you rework on it and get to the final solution

**Pull request closed without merging**